### PR TITLE
Vocabularies endpoint

### PR DESCRIPTION
**Pull request by csenger**
This is a basic implementation of #212 .

### CODE REVIEW

**Comment by sneridagh** [comment]
What about getting all the vocabularies that are available in the site?

GET /@vocabularies

/cc @tisto @csenger @lukasgraf @buchi

**Comment by tisto** [comment]
@csenger I think having a listing of all available vocabularies is quite useful. Would you mind adding that?

**Comment by sneridagh** [comment]
@tisto Just added the permissions, updated documentation format and implemented also the GET to /@vocabularies to get all the available vocabularies.

**Comment by sneridagh** [comment]
@tisto I hesitated about how to return a plain list of all the available vocabularies... Now it's a plain list. or should it be of the form this?:
```
[{"@id":"http://..../@vocabularies/plone.app.vocabularies.ReallyUserFriendlyTypes"
...
]
```

**Comment by tisto** [comment]
@sneridagh I would go with a dict with "@id" and "title" to be consistent with our other listings.

File: CHANGES.rst [hunk 1]
```diff
@@ -50,6 +50,10 @@ Bugfixes:
 
 New Features:
 
+- Add @vocabularies endpoint.
+  https://github.com/plone/plone.restapi/issues/212
+  [csenger]
```
Comment (tisto) [inline CHANGES.rst]: @sneridagh you forgot to add your name here as well. :)

File: src/plone/restapi/profiles/default/rolemap.xml [hunk 2]
```diff
@@ -3,5 +3,9 @@
     <permission name="plone.restapi: Use REST API" acquire="True">
       <role name="Anonymous"/>
     </permission>
+    <permission name="plone.restapi: Access Plone vocabularies" acquire="True">
```
Comment (tisto) [inline src/plone/restapi/profiles/default/rolemap.xml]: @sneridagh I had some doubts about the naming "access" and not "view". Though, looking at all the permissions we have in Plone there is no way to be consistent with all the inconsistency we already have...

File: src/plone/restapi/tests/test_services_vocabularies.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    def test_get_vocabulary(self):
+        response = self.api_session.get(
+            '/@vocabularies/plone.restapi.tests.test_vocabulary')
+
+        self.assertEqual(200, response.status_code)
+        response = response.json()
+        self.assertEqual(
+            response,
+            {u'@id': u'http://localhost:55001/plone/@vocabularies/''plone.restapi.tests.test_vocabulary',  # noqa
```
Comment (tisto) [inline src/plone/restapi/tests/test_services_vocabularies.py]: @sneridagh I usually try not to assert a large dict to make the tests more maintainable (our testrunner sucks at showing dict diffs). Though, I guess this is a matter of taste...

**Comment by tisto** [comment]
@sneridagh apart from the things I commented on, this looks good to me. Can be merged after fixing and rebase.

File: src/plone/restapi/tests/test_services_vocabularies.py (hunk 3 above)
Comment (tisto) [inline src/plone/restapi/tests/test_services_vocabularies.py]: @sneridagh I refactored the dict and now see the test failure trace I get:

```
    151/153 (98.7%)

Failure in test test_get_all_vocabularies (plone.restapi.tests.test_services_vocabularies.TestVocabularyEndpoint)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/timo/workspace/plone/plone.restapi/src/plone/restapi/tests/test_services_vocabularies.py", line 84, in test_get_all_vocabularies
    self.assertIn('plone.app.vocabularies.Roles', response)
  File "/usr/lib/python2.7/unittest/case.py", line 805, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 412, in fail
    raise self.failureException(msg)
AssertionError: 'plone.app.vocabularies.Roles' not found in [{u'@id': u'plone.app.vocabularies.Skins/', u'title': u'plone.app.vocabularies.Skins'}, {u'@id': u'plone.app.vocabularies.AvailableContentLanguages/', u'title': u'plone.app.vocabularies.AvailableContentLanguages'}, {u'@id': u'plone.schemaeditor.VocabulariesVocabulary/', u'title': u'plone.schemaeditor.VocabulariesVocabulary'}, {u'@id': u'plone.app.event.Weekdays/', u'title': u'plone.app.event.Weekdays'}, {u'@id': u'plone.contentrules.events/', u'title': u'plone.contentrules.events'}, {u'@id': u'plone.restapi.tests.test_vocabulary/', u'title': u'plone.restapi.tests.test_vocabulary'}, {u'@id': u'plone.app.discussion.vocabularies.CaptchaVocabulary/', u'title': u'plone.app.discussion.vocabularies.CaptchaVocabulary'}, {u'@id': u'plone.app.vocabularies.Actions/', u'title': u'plone.app.vocabularies.Actions'}, {u'@id': u'plone.app.contenttypes.migration.changed_base_classes/', u'title': u'plone.app.contenttypes.migration.changed_base_classes'}, {u'@id': u'wicked.vocabularies.BaseConfigurationsOptions/', u'title': u'wicked.vocabularies.BaseConfigurationsOptions'}, {u'@id': u'plone.app.content.ValidAddableTypes/', u'title': u'plone.app.content.ValidAddableTypes'}, {u'@id': u'plone.app.vocabularies.CommonTimezones/', u'title': u'plone.app.vocabularies.CommonTimezones'}, {u'@id': u'plone.restapi.tests.test_context_vocabulary/', u'title': u'plone.restapi.tests.test_context_vocabulary'}, {u'@id': u'plone.app.vocabularies.AvailableEditors/', u'title': u'plone.app.vocabularies.AvailableEditors'}, {u'@id': u'cmf.calendar.AvailableEventTypes/', u'title': u'cmf.calendar.AvailableEventTypes'}, {u'@id': u'plone.app.vocabularies.SupportedContentLanguages/', u'title': u'plone.app.vocabularies.SupportedContentLanguages'}, {u'@id': u'plone.app.vocabularies.Keywords/', u'title': u'plone.app.vocabularies.Keywords'}, {u'@id': u'plone.app.vocabularies.Groups/', u'title': u'plone.app.vocabularies.Groups'}, {u'@id': u'plone.app.vocabularies.AvailableTimezones/', u'title': u'plone.app.vocabularies.AvailableTimezones'}, {u'@id': u'plone.app.vocabularies.Catalog/', u'title': u'plone.app.vocabularies.Catalog'}, {u'@id': u'plone.app.discussion.vocabularies.TextTransformVocabulary/', u'title': u'plone.app.discussion.vocabularies.TextTransformVocabulary'}, {u'@id': u'wicked.vocabularies.CacheConfigurationsOptions/', u'title': u'wicked.vocabularies.CacheConfigurationsOptions'}, {u'@id': u'plone.app.vocabularies.Month/', u'title': u'plone.app.vocabularies.Month'}, {u'@id': u'plone.app.vocabularies.Weekdays/', u'title': u'plone.app.vocabularies.Weekdays'}, {u'@id': u'plone.app.vocabularies.WorkflowTransitions/', u'title': u'plone.app.vocabularies.WorkflowTransitions'}, {u'@id': u'plone.app.vocabularies.WeekdaysShort/', u'title': u'plone.app.vocabularies.WeekdaysShort'}, {u'@id': u'plone.app.contenttypes.metadatafields/', u'title': u'plone.app.contenttypes.metadatafields'}, {u'@id': u'Behaviors/', u'title': u'Behaviors'}, {u'@id': u'Interfaces/', u'title': u'Interfaces'}, {u'@id': u'plone.app.event.Timezones/', u'title': u'plone.app.event.Timezones'}, {u'@id': u'plone.app.vocabularies.WorkflowStates/', u'title': u'plone.app.vocabularies.WorkflowStates'}, {u'@id': u'plone.app.event.EventTypes/', u'title': u'plone.app.event.EventTypes'}, {u'@id': u'Fields/', u'title': u'Fields'}, {u'@id': u'plone.app.contenttypes.migration.atctypes/', u'title': u'plone.app.contenttypes.migration.atctypes'}, {u'@id': u'plone.app.vocabularies.SyndicatableFeedItems/', u'title': u'plone.app.vocabularies.SyndicatableFeedItems'}, {u'@id': u'plone.app.vocabularies.Users/', u'title': u'plone.app.vocabularies.Users'}, {u'@id': u'plone.app.event.AvailableTimezones/', u'title': u'plone.app.event.AvailableTimezones'}, {u'@id': u'plone.app.vocabularies.WeekdaysAbbr/', u'title': u'plone.app.vocabularies.WeekdaysAbbr'}, {u'@id': u'plone.app.vocabularies.UserFriendlyTypes/', u'title': u'plone.app.vocabularies.UserFriendlyTypes'}, {u'@id': u'plone.app.vocabularies.PortalTypes/', u'title': u'plone.app.vocabularies.PortalTypes'}, {u'@id': u'plone.app.controlpanel.WickedPortalTypes/', u'title': u'plone.app.controlpanel.WickedPortalTypes'}, {u'@id': u'plone.app.vocabularies.SyndicationFeedTypes/', u'title': u'plone.app.vocabularies.SyndicationFeedTypes'}, {u'@id': u'plone.app.vocabularies.Timezones/', u'title': u'plone.app.vocabularies.Timezones'}, {u'@id': u'plone.formwidget.relations.cmfcontentsearch/', u'title': u'plone.formwidget.relations.cmfcontentsearch'}, {u'@id': u'plone.app.vocabularies.ReallyUserFriendlyTypes/', u'title': u'plone.app.vocabularies.ReallyUserFriendlyTypes'}, {u'@id': u'plone.app.vocabularies.AllowableContentTypes/', u'title': u'plone.app.vocabularies.AllowableContentTypes'}, {u'@id': u'plone.app.contenttypes.migration.extendedtypes/', u'title': u'plone.app.contenttypes.migration.extendedtypes'}, {u'@id': u'plone.app.vocabularies.MonthAbbr/', u'title': u'plone.app.vocabularies.MonthAbbr'}, {u'@id': u'plone.app.vocabularies.Workflows/', u'title': u'plone.app.vocabularies.Workflows'}, {u'@id': u'Group Ids/', u'title': u'Group Ids'}, {u'@id': u'plone.app.event.SynchronizationStrategies/', u'title': u'plone.app.event.SynchronizationStrategies'}, {u'@id': u'plone.app.vocabularies.AllowedContentTypes/', u'title': u'plone.app.vocabularies.AllowedContentTypes'}, {u'@id': u'plone.app.vocabularies.Roles/', u'title': u'plone.app.vocabularies.Roles'}]

                    
  Ran 153 tests with 1 failures and 0 errors in 25.811 seconds.
```

Got my point? ;)

Comment (tisto) [inline src/plone/restapi/tests/test_services_vocabularies.py]: @csenger actually I think this is your code :)

**Comment by tisto** [comment]
I fixed the issues I raised above. PR can be merged in my opinion...

**Comment by tisto** [comment]
@lukasgraf I played around a bit with my git config but I couldn't figure out what I am doing wrong. Would you mind doing your magic again on this pr?

**Comment by lukasgraf** [comment]
@tisto sure thing 👍

**Comment by lukasgraf** [comment]
Don't currently have the time to do an in-depth review of this, but permission-wise this looks good to me! 👍

**Pull request merged**